testbed/core/utils/logging_filters.py

```diff
@@ -8,70 +8,84 @@
 
 
 def clear_trace_context():
-    # Clear trace context for the current thread/request
-    _trace_local.trace_context = None
+    _trace_context_var.set(None)
+    _request_path_var.set(None)
+    _request_method_var.set(None)
 
 
 class CloudRunTraceFilter(logging.Filter):
     """
-    Enriches log records with Cloud Run trace context and environment metadata.
+    Enrich log records with Cloud Run trace context and environment metadata.
     
     Automatically adds:
     - trace: projects/PROJECT_ID/traces/TRACE_ID (for Cloud Logging correlation)
-    - span_id: Span ID from X-Cloud-Trace-Context
-    - environment: ENV_NAME (staging/production)
+    - spanId: Span ID from X-Cloud-Trace-Context
+    - environment: ENVIRONMENT (staging/production)
     - service: K_SERVICE (Cloud Run service name)
     - revision: K_REVISION (Cloud Run revision name)
+    - request_path: HTTP request path
+    - request_method: HTTP request method
     """
     
     def __init__(self, name=""):
+        
         super().__init__(name)
         self.project_id = os.environ.get('GOOGLE_CLOUD_PROJECT', 'unknown')
-        self.environment = os.environ.get('ENV_NAME', 'unknown')
+        self.environment = os.environ.get('ENVIRONMENT', 'unknown')
         self.service = os.environ.get('K_SERVICE', 'unknown')
         self.revision = os.environ.get('K_REVISION', 'unknown')
     
     def filter(self, record):
-        # Add trace and environment context to log record
-        # Add environment metadata
-        record.environment = self.environment
-        record.service = self.service
-        record.revision = self.revision
+        if self.environment:
+            record.environment = self.environment
+        if self.service:
+            record.service = self.service
+        if self.revision:
+            record.revision = self.revision
         
-        # Add trace context if available
-        trace_context = get_trace_context()
-        if trace_context:
+        # Add trace correlation if trace header present
+        trace_context = _trace_context_var.get()
+        if trace_context and self.project_id:
             try:
-                # Parse X-Cloud-Trace-Context: TRACE_ID/SPAN_ID;o=TRACE_TRUE
-                # Convert to: projects/PROJECT_ID/traces/TRACE_ID
-                parts = trace_context.split('/')
-                if len(parts) >= 1:
-                    trace_id = parts[0]
-                    record.trace = f"projects/{self.project_id}/traces/{trace_id}"
+                # Parse X-Cloud-Trace-Context: TRACE_ID/SPAN_ID;o=1
+                # https://docs.cloud.google.com/trace/docs/trace-context
+                x_cloud_trace_context_parts = trace_context.split(";")[0].split("/")
+                if len(x_cloud_trace_context_parts) >= 1:
+                    trace_id = x_cloud_trace_context_parts[0]
+                    if trace_id:
+                        record.trace = f"projects/{self.project_id}/traces/{trace_id}"
```

Comment:
If the record is being passed by value instead of by reference, then changes to the object wouldn't be carried along.  Are we sure that this is an intent of the logging.Filter interface?  It seems more like taking advantage of a (possibly temporary or situational) side effect of the interface.